### PR TITLE
Use translated Hint instead of english error on password policy

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by nickvergessen** [comment]
backport in https://github.com/nextcloud/server/pull/5434

File: settings/Controller/UsersController.php [hunk 1]
```diff
@@ -406,6 +407,9 @@ public function create($username, $password, array $groups=[], $email='') {
 			$user = $this->userManager->createUser($username, $password);
 		} catch (\Exception $exception) {
 			$message = $exception->getMessage();
+			if ($exception instanceof HintException && $exception->getHint()) {
+				$message = $exception->getHint();
+			}
 			if (!$message) {
```
Comment (blizzz) [inline settings/Controller/UsersController.php]: ``else`` should be enough since you factually check for $message in the previous if condition

Comment (MorrisJobke) [inline settings/Controller/UsersController.php]: Unrelated to this PR ;)

Comment (blizzz) [inline settings/Controller/UsersController.php]: If you touch this line already… but whatever, should not block it.

**Review by blizzz — APPROVED** [review]

**Pull request merged**